src/logics/Logic.cc

```diff
@@ -7,1632 +7,1657 @@
-    auto name = isQuoted(protectedName) ?
-            std::string_view(protectedName.data()+1, protectedName.size()-2) : std::string_view(protectedName);
-
-    if (not isKnownToUser(name) or isAmbiguousUninterpretedNullarySymbolName(name)) {
-        return "(as " + std::string(protectedName) + " " + printSort(sortRef) + ')';
-    } else {
-        return protectedName;
-    }
-}
-
-//
-// Quote the name if it contains illegal characters
-//
-std::string Logic::protectName(std::string const & name, bool isInterpreted) const {
-    assert(not name.empty());
-    if (not isInterpreted and (hasQuotableChars(name) or std::isdigit(name[0]) or isReservedWord(name))) {
-        return '|' + name + '|';
-    }
-    return name;
-}
-
-// Return a string corresponding to the SMT lib representation of the symbol, with disambiguation and name protection
-std::string Logic::printSym(SymRef sr) const {
-    Symbol const & symbol = getSym(sr);
-    bool isInterpreted = symbol.isInterpreted();
-    std::string protectedName = protectName(getSymName(sr), isInterpreted);
-    return disambiguateName(std::move(protectedName), getSortRef(sr), symbol.nargs() == 0, isInterpreted);
-}
-
-
-std::string Logic::pp(PTRef tr) const {
-    const Pterm &t = getPterm(tr);
-    SymRef sr = t.symb();
-    std::string name_escaped = printSym(sr);
-
-    if (t.size() == 0) {
-        return name_escaped;
-    }
-
-    assert(t.size() > 0);
-
-    std::stringstream ss;
-    ss << '(' << name_escaped << ' ';
-    for (int i = 0; i < t.size(); i++) {
-        const std::string arg = pp(t[i]);
-        ss << arg;
-        if (i < t.size()-1) {
-            ss << ' ';
+namespace opensmt {
+    using namespace std;
+
+    /***********************************************************
+     * Class defining logic
+     ***********************************************************/
+
+    char const * Logic::e_argnum_mismatch = "incorrect number of arguments";
+    char const * Logic::e_bad_constant = "incorrect constant for logic";
+
+    char const * Logic::tk_val_uf_default = "UFDefault";
+    char const * Logic::tk_val_bool_default = "true";
+
+    char const * Logic::tk_true = "true";
+    char const * Logic::tk_false = "false";
+    char const * Logic::tk_not = "not";
+    char const * Logic::tk_uf_not = ".uf-not";
+    char const * Logic::tk_equals = "=";
+    char const * Logic::tk_implies = "=>";
+    char const * Logic::tk_and = "and";
+    char const * Logic::tk_or = "or";
+    char const * Logic::tk_xor = "xor";
+    char const * Logic::tk_distinct = "distinct";
+    char const * Logic::tk_ite = "ite";
+    char const * Logic::tk_indexed = "_";
+
+    char const * Logic::s_sort_bool = "Bool";
+    char const * Logic::s_framev_prefix = ".frame";
+    char const * Logic::s_abstract_value_prefix = "@";
+
+    // The constructor initiates the base logic (Boolean)
+    Logic::Logic(Logic_t _logicType)
```

Comment:
This Indentation is used in all formatted files. I only did not bother about files that are not formatted, and also to keep the number of modified lines lower.
This kind of Indentation was already defined in `.clang-format` before. I think it was me who introduced it, but back then I probably thought it was ok because you did not complain (?)
Do you have arguments against it?

Reply:
I strongly prefer that namespace does not increase indentation level.
If you are writing a function in an anonymous namespace inside another namespace, the indentation can get quite annoying, and it also limits the number of character per line you have for the actual code.

Reply:
Fair enough. However, in the case of nested namespaces (i.e. not the outmost one), I feel like it is good to emphasize that nested namespace is entered. These often provide implementation details or "hacks", which is a good thing to spot. This would correspond to `NamespaceIndentation: Inner` in clang-format. I also think that at most one nested namespace will appear, because except of anonymous ones, it is usually better to use c++17 grouped namespaces instead if nested (I.e. `namespace a::b {}`.

Reply:
Sounds like a good compromise.

Reply:
I take the "hacks" back, reimplementing functions in anonymous namespaces does not compile because it introduces ambiguity..

Reply:
> Sounds like a good compromise.

See `src/tsolvers/lasolver/FarkasInterpolator.cc` where there is quite a long anonymous namespace. Check if it looks well for you.